ArduCopter/fence.cpp

```diff
@@ -31,8 +33,18 @@ void Copter::fence_check()
             }else{
                 // if we are within 100m of the fence, RTL
                 if (fence.get_breach_distance(new_breaches) <= AC_FENCE_GIVE_UP_DISTANCE) {
-                    if (!set_mode(RTL, MODE_REASON_FENCE_BREACH)) {
-                        set_mode(LAND, MODE_REASON_FENCE_BREACH);
+                    if (fence_act == AC_FENCE_ACTION_BRAKE) {
+                        if (!set_mode(BRAKE, MODE_REASON_FENCE_BREACH)) {
+                            set_mode(LAND, MODE_REASON_FENCE_BREACH);
+                        }
+                    } else if (fence_act == AC_FENCE_ACTION_SMART_RTL) {
+                        if (!set_mode(SMART_RTL, MODE_REASON_FENCE_BREACH) && !set_mode(RTL, MODE_REASON_FENCE_BREACH)) {
```

Comment:
I don't think this is good, as it will go to smartRTL and then directly to RTL, isn't it ?

Reply:
right , I miss the inverted logic.... so it works.